rtl/include/eclhelper.hpp

```diff
@@ -363,6 +364,7 @@ struct RtlTypeInfo : public RtlITypeInfo
     inline bool isFixedSize() const { return (fieldType & RFTMunknownsize) == 0; }
     inline bool isLinkCounted() const { return (fieldType & RFTMlinkcounted) != 0; }
     inline bool isUnsigned() const { return (fieldType & RFTMunsigned) != 0; }
+    inline bool xpathIsScalar() const { return (fieldType & RFTMxpathscalar) != 0; }
```

Comment:
This function name is slightly confusing, since it will return true for xpaths on non-scalar fiels.   isValueScalarXPath() is closer.  It may seem a bit picky, but it makes it much easier to read the code later if it is accurate.